enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,360 @@
+- Develop a client interface that facilitates easy integration for consumers.
+
+### Non-Goals
+
+- Define the controlplane to act as a consumer of the protocol.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+- As a hub controller developer, I can seamlessly transition to event-based manifestworks without modifying the controller logic by switching from the Kubernetes-style approach.
+
+#### Story 2
+
+- As a service developer, I can effortlessly utilize the event-based manifestwork API to construct a service that delivers manifestworks to managed clusters.
+
+### Risks and Mitigation
+
+- The authentication to the event transport layer (TBD)
+
+## Design Details
+
+We will adopt the CloudEvents specification as the message format for ManifestWorks. Multiple API consumers(sources) can leverage this format, and each consumer(source) must have a unique source ID to ensure proper identification and handling of events.
+
+To ensure uniqueness and immutability of source components, each source, whether it is a consumer on the open cluster management hub cluster or a restful service handling resource requests, should possess a unique identifier. For instance, the open cluster management hub consumer can generate a source ID by hashing the hub cluster URL and appending a consumer name. Similarly, a restful service can choose a unique name or generate a unique ID in the associated database for its source identification.
+
+### Spec Event
+
+Sources send their resources along with the message to the work agent, enabling the creation, updating, or deletion of the respective resources on the managed cluster.
+
+Event transfer flow:
+
+1. The work agent subscribes to the spec event topic `/sources/+/clusters/<cluster-id>/manifests` from the broker, for example: `/sources/+/clusters/cluster1/manifests`.
+2. The source receives high level API request and publishes its resources using the spec event to the topic `/sources/<source-id>/clusters/<cluster-id>/manifests`, for example: `/sources/sd3ded4-mwrs-controller/clusters/cluster1/manifests`.
+3. The work agent receives the published event from the source and deploys the manifests included in the event.
+
+```mermaid
+sequenceDiagram
+    WorkAgent -->> Broker: Subscribe topic - /sources/+/clusters/<cluster-id>/manifets
+    Source -->> Source: Receive high-level API request and encode manifestwork into spec event
+    Source -->> Broker: Publish spec event for manifestwork
+    Broker -->> WorkAgent: Send spec event for manifestwork
+    WorkAgent -->> WorkAgent: Decode spec event and apply manifestwork
+```
+
+Event Schema:
+
+```json
+{
+    "specversion": "1.0", //The version of the CloudEvents specification which the event uses.
+    "id": "<cloud-events-message-id>",
+    "source": "<source-id>", //Identifies the context in which an event happened. Often this will include information such as the type of the event source, the organization publishing the event or the process that produced the event.
```

Comment:
let's have an exact definition for source filed, I think it should be message producer topic, e.g.
`/sources/sd3ded4-mwrs-hub-controller/clusters/cluster1/manifests`
`/sources/sd3ded4-mwrs-agent-controller/clusters/cluster1/manifestsstatus`

Reply:
done.